gesticulator/data_processing/text_features/parse_json_transcript.py

```diff
@@ -87,10 +86,14 @@ def encode_json_transcript_with_bert(json_file, tokenizer, bert_model):
             elif word[:-1] in fillers: # The last character of the word might be a delimiter
                 curr_word_idx = 1
 
-                # Here we explicitly check whether the delimiter signals the end of the sentence
-                # For example, commas are not added to the word list
+                # If filler is the last word in the sentence, then we add its delimiter to the last word in the sentence
                 if word[-1] in delimiters: 
-                    non_filler_words_in_sentence.append(word[-1])
+                    if non_filler_words_in_sentence != []:
+                        # Append to the last word
+                        non_filler_words_in_sentence[-1] += word[-1]
+                    else:
+                        # Add the delimiter as the only word in the sentence
+                        non_filler_words_in_sentence.append(word[-1])
```

Comment:
this if was added in case the sentence is empty (e.g. "Ah." will be ["."] since "Ah" is ignored"